### ISSUE TITLE
[BUG][Block 656398]  `OGD: Invalid signature`

### ISSUE THREAD

**Issue by pefontana** [lambdaclass/cairo_native#740]
When executing Block 656398 with starknet-replay, we get this error in certain transactions 
```
Failure reason: 0x4f47443a20496e76616c6964207369676e6174757265 ('OGD: Invalid signature') or 
Native execution error: OGD: Invalid signature\n"
```

transactions hash that triggers the error:
* 0x2a23e4131787cc48259d25ea7940c2b65c2f2ca81532cd5df3fc5a0ffcd8ec5
* 0x4d4380391a891824a8cfaecb5f70da151034c52c376d92385c1597bb5b62c90
* 0x176260cd079139d2c1a2afed5cf2b1c945059f7149f3ddab9b38cfcb8662050
* 0x763231585a90c12e4a020f1d8532063bea1935254b36c605f87a5661dfea73a
* 0x7a9d5725a87065770127ab450d4acefb0fa6fcd27b180261de9dcda823c3c54

Also in block 619888
transaction_hash: "0x7f18e97fc92f3939098ea3ed57b9ca6062bd5aba7e4eb5ead0e32bb580b3e80"

PD. Check what happen when you execute with the cairo-vm

**Comment by ShantelPeters** [comment]
Hi @pefontana i will love to work on this

**Comment by FrancoGiachetta** [comment]
Update

Using the main branch of starknet-replay (llvm 18)
 
Running again the block 656398 with both native and vm, resulted as flows:
- Native: did not finished the execution. Failed executing transaction `0x6a09079fbc6e1fe092d61aeec83e33c4bc2900f7bc311cf081cbbd34d3d2be6` with message `Killed`.
- VM: The transactions failed with the same error message (`OGD: Invalid signature`)

Running again the block 619888 with both native and vm, resulted as flows:
tx `0x7f18e97fc92f3939098ea3ed57b9ca6062bd5aba7e4eb5ead0e32bb580b3e80` returns `OGD: Invalid signature\n` with both the vm and native

**Comment by pefontana** [comment]
Fixed with #787 and #763
